src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -67,15 +74,31 @@ public void deleteUser(
 
     @PutMapping("/{id}")
     @ApiOperation(value = "Updates an user's information")
-    public User updateUser(@RequestBody User user, @PathVariable Long id)
+    public User updateUser(@RequestBody User dto, @PathVariable Long id)
         throws UserMismatchException, UserNotFoundException {
-        if (user.getId() != id) {
+        if (dto.getId() != id) {
             throw new UserMismatchException();
         }
-        userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        // Update all but password
+        user.setBirthdate(dto.getBirthdate());
+        user.setName(dto.getName());
+        user.setUsername(dto.getUsername());
         return userRepository.save(user);
     }
 
+    @PutMapping("/{id}/password")
+    @ApiOperation(value = "Updates an user's password")
+    public User updatePassword(@RequestBody User dto, @PathVariable Long id)
+        throws UserNotFoundException, UserMismatchException {
+        if (dto.getId() != id) {
+            throw new UserMismatchException();
+        }
+        User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        user.setPassword(passwordEncoder.encode(dto.getPassword()));
```

Comment:
The password encoding logic fits better in the model, not the controller (the method `setPassword` should include the encryption)